src/appleseed/renderer/modeling/camera/thinlenscamera.cpp

```diff
@@ -735,20 +737,34 @@ DictionaryArray ThinLensCameraFactory::get_input_metadata() const
             .insert("use", "required")
             .insert("default", "8.0"));
 
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "autofocus_enabled")
+            .insert("label", "Enable autofocus")
+            .insert("type", "boolean")
+            .insert("use", "optional")
+            .insert("default", "true"));
+
     metadata.push_back(
         Dictionary()
             .insert("name", "focal_distance")
             .insert("label", "Focal Distance")
             .insert("type", "text")
-            .insert("use", "optional"));
+            .insert("use", "optional")
+            .insert("visible_if",
+                Dictionary()
+                    .insert("autofocus_enabled", "false")));
```

Comment:
I think input box appears/disappears interactively, but it doesn't seems good : for example, `focal_distance` checkbox appears after I uncheck `autofocus_enabled` -> press 'OK' -> and reopen edit window. It was quite inconvenient for me.

Reply:
It's definitely supposed to show/hide widgets immediately, without having to close and reopen the window. You can see this in action by editing the frame entity (Output section in the Project Explorer) and changing the value in the Denoiser dropdown.

Reply:
Would appreciate further investigating what's going on :)

Reply:
Then how about delete contition for now? I can try to fix it later.

Reply:
I agree, let's do this in a separate PR.